codec-redis/src/main/java/io/netty/handler/codec/redis/StringRedisMessage.java

```diff
@@ -0,0 +1,65 @@
+/*
+ * Copyright 2016 The Netty Project
+ *
+ * The Netty Project licenses this file to you under the Apache License, version 2.0 (the
+ * "License"); you may not use this file except in compliance with the License. You may obtain a
+ * copy of the License at:
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package io.netty.handler.codec.redis;
+
+import io.netty.util.internal.ObjectUtil;
+import io.netty.util.internal.StringUtil;
+
+/**
+ * Simple Strings or Errors of <a href="http://redis.io/topics/protocol">RESP</a>.
+ */
+public class StringRedisMessage implements RedisMessage {
+
+    private final RedisMessageType type;
+    private final String content;
+
+    StringRedisMessage(RedisMessageType type, String content) {
+        ObjectUtil.checkNotNull(content, "content");
+        if (type != RedisMessageType.SIMPLE_STRING && type != RedisMessageType.ERROR) {
```

Comment:
instead of passing in a `RedisMessageType` what about passing in a `boolean isSimpleString` where `true` means `type = RedisMessageType.SIMPLE_STRING` and `false` means `type = RedisMessageType.ERROR`. This way there is no need for "invalid" check here.